modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/MergeJoinConverterRule.java

```diff
@@ -51,7 +52,13 @@ public MergeJoinConverterRule() {
     @Override public boolean matchesJoin(RelOptRuleCall call) {
         LogicalJoin logicalJoin = call.rel(0);
 
-        return !F.isEmpty(logicalJoin.analyzeCondition().pairs()) && logicalJoin.analyzeCondition().isEqui();
+        /**
+         * TODO might be revised after taking https://issues.apache.org/jira/browse/CALCITE-6927
```

Comment:
1. TODO should not contains fixed ticked.
2. We don't need to expand IS_NOT_DISTINCT_FROM, since know how to deal with it. So, no TODO needed.